Web Dev/FrontEnd/Libraries & Frameworks/React JS .md

```diff
@@ -0,0 +1,257 @@
+
+- Flexibility: React can be used to build a wide range of applications, including single-page applications, mobile apps, and desktop apps.
+
+- Large Community: React has a large and active community, which means that you can find answers to your questions and get support from other developers.
+## 🎓 Take Away Skills
+
+After completing this learning path you will be knowledgeable in:
+ - Building web applications
+ - React Components
+ - JSX
+ - State Management
+ - React Router
+ - Redux
+ - Testing
+
+---
+## 🛠️ Prerequisites
+
+## 🧑🏻‍💻 Programming Knowledge 
+ 
+- JavaScript: React is built on top of JavaScript, so having a good understanding of JavaScript fundamentals is essential. You should be comfortable with concepts like variables, functions, arrays, objects, loops, and conditional statements.
+
+- HTML: React is used to build user interfaces, so you should have a good understanding of HTML markup.
+
+- CSS: React also requires a good understanding of CSS styling and layout, as you'll be using it to style your components.
+
+you can learn [javascript here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide) , [HTML here](https://www.codecademy.com/learn/learn-html) , [CSS here](https://www.freecodecamp.org/learn/responsive-web-design/#basic-css)
+
+## 📲 Installation and Setup
+
+You need a code editor,browser and **Node.js** to follow along this learning path. You can choose the code editor and browser of your choice. The preferred ones are listed below:
+- **VS Code**
+
+    Light-weight code editor by Microsoft with a large ecosystem of plugins to help your workflow
+    
+    [Install VS Code ](https://code.visualstudio.com/download)
+- **Google Chrome**
+    
+    A browser based on V8 JavaScript engine with developer tools.
+    
+    [Install Google Chrome](https://www.google.com/chrome/)
+- **Node.js**
+
+    JavaScript runtime which helps you run JavaScript programs in your system.
+
+    [Install Node](https://nodejs.org/en/download/)
+
+    ***Note:*** Install the stable version
+---
+
+## 💡 Learning Session
+
+### Rapid Review on JS
+
+**🎓 Topics to Learn**
+```
+- Variables and data types
+- Functions
+- Conditionals and loops
+- Arrays and objects
+- DOM manipulation
+- Asynchronous JavaScript (AJAX, callbacks, promises)
+```
+
+**🧑🏻‍💻 Learn from**
+
+
+*Website Tutorials*
+- [W3 schools](https://www.w3schools.com/js/)
+
+*Video Tutorials*
+- [JavaScript Required for React JS](https://youtu.be/m55PTVUrlnA)
+- [JavaScript Required for learning React](https://youtu.be/WzXaeRQKI0c)
+
+*Articles/Blogs*
+
+- [JavaScript Required to learn React JS](https://blog.logrocket.com/javascript-concepts-before-learning-react/)
+
+**🛠️ Get into action**
+- Both of the above mentioned resources include a lot of activities. Try those activities to get a better understanding of JavaScript
+
+### Fundamentals in React
+
+**🎓 Topics to Learn**
+```
+- Create React App

```

Comment:
it would be better to add vite or next js here ,  new react docs dont recomend create-react-app anymore ! (not manditory though)